SecretSanta/src/SecretSanta.Api/Controllers/BaseApiController.cs

```diff
@@ -35,13 +37,19 @@ public async Task<IActionResult> Get(int id)
         }
 
         [HttpPut("{id}")]
-        public async Task<TEntity?> Put(int id, [FromBody] TEntity value)
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> Put(int id, [FromBody] TInputDto value)
         {
-            return await Service.UpdateAsync(id, value);
+            if (await Service.UpdateAsync(id, value) is Gift entity)
```

Comment:
Given that this is in the base class, `Gift` is not what you want here.
```suggestion
            if (await Service.UpdateAsync(id, value) is TDto entity)
```